### ISSUE TITLE
On creating a unified organization & publishing

### ISSUE THREAD

**Issue by jorgepz** [MOxUnit/MOxUnit#88]
Hi @Moxunit and @MOcov developers, I find MOxunit and Mocov very useful tools with great potential as software testing becomes more and more essential. I have two comments/questions:

1- have you considered publishing them, for instance in joss? I would guess that the magnitude of work you have done and the usability of the tools would justify a publication there.

2- have you considered creating a unified organization with moxunit, [mocov
(https://github.com/MOcov/MOcov), a website, maybe a moxunit-action such as [this one](https://github.com/joergbrech/moxunit-action) by @joergbrech and maybe including tutorials/notebooks or minimal working examples such as [this one](https://github.com/jorgepz/Octave_Matlab_CI_Testbed) to help new users to adopt the tools?

**Comment by nno** [comment]
Dear @jorgepz, thanks for your kind message. It’s nice to hear that you find MOxUnit and MOcov useful :-). 

1. It’s not something I considered as I was not aware of this journal. I did have a quick look and it might actually be something worth considering. From what I saw the publications are not very long, in other words preparing a manuscript does not seem like a whole lot of work. Do you have personal experiences with this journal? 
2. Also a unified organization is not something I considered: I wrote most of the code quite a few years ago, mainly in order to test another project, CoSMoMVPA, which does have its own website ([cosmomvpa.org](http://cosmomvpa.org/)). Since then, due to a change in professional activities, I have mostly been developing in Python - my activities on the Matlab / GNU Octave platforms have been very limited. My impression so far was that there was little interest from developers, thus spending time on a unified organization or website did not seem worthwhile. But maybe I’m wrong about this. Do you have indications that there is a broader interest, besides yourself and @joergbrech (who indeed created a nice moxunit-action) in MOxUnit and MOcov --- and possibly MOdox for documentation testing?

best, Nick

**Comment by jorgepz** [comment]
Hi Nick, I am glad that these topics seem interesting.

on 1.
I haven't submitted to JOSS but I had a great experience submitting to a juliacon proceeding which (I understand) is managed using the same procedures and/or infrastructure https://github.com/JuliaCon/proceedings-review/issues/95 I totally recommend submitting to JOSS since it seems to have an open review process and would potentiate the tools. (I would suggest deciding on creating an organization before the potential submission.

on 2.
I think that even if we are just a handful of users using Moxunit et al, I think it makes sense (for many reasons) to centralize the repos in one organization. In addition, this could make the tools developed more visible to octave/matlab users (who are not so naturally related to devops concepts as in python). In my case I gradually discovered moxunit, moxunitaction, mocov and modox, and I got into devops. I think that devops will be (or is) integrated to any programming language, and this potential organization could be the centralizer of nice devops tools for matlab/octave. In the first programming courses of the school of Engineering I teach in Uruguay octave is the tool used. Having public documented centralized devops tools would be interesting for teachers to present these topics.

Best,
Jorge

**Comment by joergbrech** [comment]
I developed the action back then as part of my teaching activities. I have moved on to other projects and don't actively use it anymore *(it is more of a toy side project)*. I am totally ok with moving moxunit-action to a dedicated organization and "transferring the ownership" to the open source community and thus anybody who is more invested then me. 

I am also open for a publication in JOSS if you think it stands a chance. Let me know if and how I can contribute.

**Comment by Remi-Gau** [comment]
My 2-cents: moving to an organization makes sense also in terms of bus factor as it may help, manage maintenance across repos by adding new contributors / maintainers so a single orga rather than several different repos.

I think that JOSS stands a chance: finding reviewers for matlab things there may be a bit harder but given the clear Octave support I think it will be easier.

**Comment by jorgepz** [comment]
I agree with @Remi-Gau on the bus factor reason. On the reviewers requirements, it seems that in JOSS there are 22 published libraries with Matlab code https://joss.theoj.org/papers/in/MATLAB thus, it might be possible. there are also other non-joss matlab github users that can be suggested.

**Comment by nno** [comment]
Hello @jorgepz, @Remi-Gau, @joergbrech,

thanks for your reactions, again it’s nice to see there is interest in the project.

1. I’m in a somewhat similar situation as @joergbrech in that I don’t use MOxUnit much myself anymore. To the extent that I use it, it has most of the features I need. (Although: updated continuous integration solutions could be improved)
1. I do feel responsibility towards other users to facilitate that the project remains useful. 
1. The bus factor is a fair point (which would involve me personally!): indeed this makes the project vulnerable. Since the project is open source it could always be forked, but I would consider that — in general for open source projects, and also specifically in this case — a last resort option. 
1. I’m definitely open to the idea of sharing contributor / maintainer rights with others — it would probably make my life easier and deal with the bus factor at the same time. It would require some level of enthusiasm, responsibility, and ‘taste’ for good design. I think the bar could be reasonably set at a level such as “has made useful contribution(s) (Pull Request) in the past”, “likely not to mess up the project”, and “understands that with great power comes great responsibilities). 

That brings me to the question: what would be a good plan to go forward? Which issues should to be addressed, and in what order? As a starting list: 
- see who would be interested in becoming a co-maintainer or even lead maintainer (and spend time making actual contributions). 
- bring the different repos under one organization (*). 
- add documentation.
- host the documentation somewhere. Github pages might be a simple way to start; more advanced (but also not for free) would be through a domain name and web hosting.
- write a paper and submit to JOSS. (possibly after the points above have been addressed?)

(*) One approach would be to bring the MOcov and MOdox repos under the github MOxUnit organization. Documentation could be included in the current MOxUnit repo, or under a new repo under the MOxUnit organization. MOxUnit action (and possibly other actions) could also go here. It may also be considered to move MOdox and MOcov code into the MOxUnit repository. 
@joergbrech, thanks for being willing to transfer ownership of moxunit-action under one organization, that's much appreciated. 

As I mentioned earlier: my involvement in the project is fairly minimal at the moment, so I think the above would be worthwhile and realistic only if other people would be wiling to do a considerable part of the work in a contributor / maintainer role. 

I’m looking forward to hearing your thoughts. 

Regarding a contributor / maintainer role: anyone interested? I’d be happy to meet in an online video call, if anyone would like to discuss this with me. 

best,
Nick

**Comment by jorgepz** [comment]
great to see major contributors aligned! thanks also for your disposition @nno

indeed I think that at this point a call is the way to go, we can coordinate via email mine in jorgepz@fing.edu.uy .

I am a very minor contributor of MoCov and user of moxunit-action. I have disposition and interest to become a more active contributor and in a near future, if needed and the planets aligned, a maintainer.

I have some experience since I'm the owner of ONSAS organization. The documentation of https://github.com/ONSAS/ONSAS used to be published via github pages using actions for free and without paying a domain. pages are free for open orgs. Paying a domain would require another lever of effort managing money... I do not disapprove the idea but I think it is best to see how the org works before and also publish in JOSS. 

In terms of movements: bringing modox and mocov to moxunit (using the transfer option in github) seems the best, and maybe changing the name of the org... i waas going to propose MODO: Matlab Octave Dev Ops , but it's already taken https://github.com/modo 🤣  ... or just leave moxunit ...

**Comment by Remi-Gau** [comment]
Agree that a quick call may help: remi.gau2@mcgill.ca

- see who would be interested in becoming a co-maintainer

I could help with some maintenance. I have not seen a lot of traffic on those repos, so I don't expect this to be a time sink (famous last words)

- bring the different repos under one organization (*).

MOxUnit organization sounds good

- add documentation & host the documentation somewhere

github pages or read the docs for hosting are pretty easy when using sphinx or mkdocs

**Comment by nno** [comment]
Thanks for the replies. @Remi-Gau I sent you a personal message regarding a call. If anyone else would like to join please let me know!

**Comment by wahln** [comment]
Hi everybody,
I started using MOxUnit and MOcov a while ago for a [MATLAB-based project](https://github.com/e0404/matRad) I am currently maintaining (in the field of Medical Physics). I've found MOxUnit and MOcov extremely useful for CI there, due to the Octave compatibility we want to test against. I've made a few minor edits to MOcov and MOxUnit for that purpose and opened a few PRs to MOcov (e.g. [here](https://github.com/MOxUnit/MOcov/pull/32) and [here](https://github.com/MOxUnit/MOcov/pull/50)).

If desired, I would be willing to invest a bit of time on PR reviewing, documentation, etc. 
For documentation, I alsp have experience with mkdocs and sphinx publishing on readthedocs, for example, as well as github pages.

Let me know if you'd like any help.
